lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -380,7 +380,7 @@ public function update_item( $request ) {
 		}
 
 		if ( ! empty( $request['slug'] ) && $request['slug'] !== $user->user_nicename && get_user_by( 'slug', $request['slug'] ) ) {
-			return new WP_Error( 'rest_user_invalid_slug', __( 'Slug is invalid.' ), array( 'status' => 400 ) );
+			return new WP_Error( 'rest_user_invalid_slug', __( 'Slug is invalid.' ), array( 'status' => 404 ) );
```

Comment:
This should remain a `400` error because the provided `slug` is not allowed and the request needs to be modified.